library/alloc/src/collections/btree/node.rs

```diff
@@ -118,96 +118,77 @@ struct BoxedNode<K, V> {
 }
 
 impl<K, V> BoxedNode<K, V> {
-    fn from_leaf(node: Box<LeafNode<K, V>>) -> Self {
-        BoxedNode { ptr: Unique::from(Box::leak(node)) }
-    }
-
-    fn from_internal(node: Box<InternalNode<K, V>>) -> Self {
-        BoxedNode { ptr: Unique::from(Box::leak(node)).cast() }
+    fn from_owned(node: NonNull<LeafNode<K, V>>) -> Self {
+        BoxedNode { ptr: unsafe { Unique::new_unchecked(node.as_ptr()) } }
     }
 
     fn as_ptr(&self) -> NonNull<LeafNode<K, V>> {
         NonNull::from(self.ptr)
     }
 }
 
-/// An owned tree.
+/// An owned tree. Also, the volatile representation of the contents of an internal edge,
```

Comment:
I think volatile here is likely misleading -- there's no "volatile" in the read_volatile/write_volatile sense here.
Maybe something like "temporary"" would be better? But it also seems like it's maybe not hugely necessary to specify, either.

Reply:
I hadn't heard of that kind of volatile in Rust, and temporary does sound better, but I don't remember myself what the point was I was trying to make, so I'll just revert it.